sklearn/utils/_print_versions.py

```diff
@@ -0,0 +1,118 @@
+"""
+Utility methods to print system info for debugging
+
+adapted from :func:`pandas.show_versions`
+"""
+# License: BSD 3 clause
+
+import platform
+import sys
+import importlib
+
+from .._build_utils import get_blas_info
+
+
+def _get_sys_info():
+    """System information
+
+    Return
+    ------
+    sys_info : dict
+        system and Python version information
+
+    """
+    python = sys.version.replace('\n', ' ')
+
+    blob = [
+        ("python", python),
+        ('executable', sys.executable),
+        ("machine", platform.platform()),
+    ]
+
+    return dict(blob)
+
+
+def _get_deps_info():
+    """Overview of the installed version of main dependencies
+
+    Returns
+    -------
+    deps_info: dict
+        version information on relevant Python libraries
+
+    """
+    deps = [
+        # (MODULE_NAME, f(mod) -> mod version)
+        ("pip", lambda mod: mod.__version__),
+        ("setuptools", lambda mod: mod.__version__),
+        ("sklearn", lambda mod: mod.__version__),
+        ("numpy", lambda mod: mod.__version__),
+        ("scipy", lambda mod: mod.version.version),
+        ("Cython", lambda mod: mod.__version__),
+        ("pandas", lambda mod: mod.__version__),
+        ("matplotlib", lambda mod: mod.__version__),
```

Comment:
As previously mentioned it might be worth removing matplotlib. I don't think it's worth an additional line in each report. For the very few issues that do need the matplotlib version, it can be asked in comments.